### PR TITLE
Add /not-found page

### PR DESCRIPTION
**Pull request by mabelzhang**
Part of #16 (the easy part :sweat_smile: Bear with me as I learn about how the website works :sweat_smile: )

This creates the `/not-found` page, which is referred to twice by the doc page `doc.component.ts` [here](https://github.com/gazebo-web/gazebosim-web-frontend/blob/4da2478c9ed82db0ae0c3f21bb70627d6a7a91ea/src/app/doc/doc.component.ts#L207) and [here](https://github.com/gazebo-web/gazebosim-web-frontend/blob/4da2478c9ed82db0ae0c3f21bb70627d6a7a91ea/src/app/doc/doc.component.ts#L234)

This one-line page probably doesn't need a big CSS, but... I just copied from `support.component.scss`,which is an equally empty page :woman_shrugging: 

## To test

I used the `docker_dev` branch in both gazebosim-web-frontend and gazebosim-web-backend repos to test it (merge to that branch locally):
```
cd path/to/gazeobsim-web-frontend
docker build . -t gazebosim_web_frontend
cd path/to/gazebosim-web-backend
docker build . -t gazebosim_web_backend
cd path/to/gazeobsim-web-frontend
docker-compose up
```

Navigate to http://localhost:3000/not-found , and it should say "The page is not found." (as opposed to just showing the home page, which is the old behavior)
A doc URL with the distro name e.g. http://localhost:3000/docs/harmonic/bad_url should redirect to the `not-found` page (this is the second part of #16).

One small annoyance is that it appears to be a redirect, so if you hit the back button in the browser, you're just redirected back to the same page. You have to go back 2 pages to get back to the previous page.

(This PR does not address the first part of #16, which is for URLs without the distro name, lke http://localhost:3000/docs/bad_url )

I can't assign reviewers so tagging manually @azeey

### CODE REVIEW

File: src/app/not-found/not-found.component.scss
Comment (azeey) [inline src/app/not-found/not-found.component.scss:1]: Removing this file didn't seem to do anything, so I don't think it's needed

File: src/app/not-found/not-found.component.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Component, OnInit } from '@angular/core';
+import { Meta, Title } from '@angular/platform-browser';
+
+@Component({
+  selector: 'gz-not-found',
+  templateUrl: 'not-found.component.html',
+  styleUrls: ['not-found.component.scss']
```
Comment (azeey) [inline src/app/not-found/not-found.component.ts]: I think we can remove this

**Review by azeey — APPROVED** [review]
Seems to work for the case you mentioned (docs/<release>/bad_url). This is how it looks locally.

Screenshot 2023-08-24 22 57 40

I just have a couple minor comments.

**Comment by mabelzhang** [comment]
Should be addressed now

**Comment by mabelzhang** [comment]
I don't have permission to merge :)

**Pull request merged**